test/array_test.cpp

```diff
@@ -5,7 +5,7 @@
 BOOST_AUTO_TEST_CASE(size_test)
 {
   rabbit::value v;
-  BOOST_CHECK_THROW(v.size(), rabbit::type_mismatch);
+  BOOST_CHECK_THROW(v.size(), std::runtime_error);
```

Comment:
Why change type_mismatch to runtime_error?

Reply:
I felt as though it wasn't a properly descriptive exception type anymore. To me mismatch implies that there is only 1 type that would actually match, when in fact now there are two types that could be valid here. I'm not too attached to it to not make changes though, I can prep a change back to type_mismatch.

Reply:
I also feel that the name type_mismatch is not good. But please use this currently.